src/library_noderawfs.js

```diff
@@ -122,10 +122,30 @@ mergeInto(LibraryManager.library, {
     allocate: function() {
       throw new FS.ErrnoError(ERRNO_CODES.EOPNOTSUPP);
     },
-    mmap: function() {
-      throw new FS.ErrnoError(ERRNO_CODES.ENODEV);
+    mmap: function(stream, address, length, position, prot, flags) {
+      if (stream.stream_ops) {
+        // this stream is created by in-memory filesystem
+        return VFS.mmap(stream, address, length, position, prot, flags);
+      }
+
+      // We don't currently support location hints for the address of the mapping
+      assert(address === 0);
```

Comment:
Return errno here rather than asserting?

Reply:
Do you know which errno is appropriate in this case? `ENXIO` or just `ENODEV`? Note that we need to change `library_nodefs.js` as well:
https://github.com/emscripten-core/emscripten/blob/097cfb8726001cf81343f1a9f61e06b5074ace85/src/library_nodefs.js#L299-L300

Reply:
EINVAL seems reasonable enough.

Reply:
Done with commit https://github.com/kleisauke/emscripten/commit/6ac5c096b01875b7e2c10d55a15266e346bacdec, thanks!